module/polls/helper.js

```diff
@@ -0,0 +1,526 @@
+                         }
+                    });
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_CREATION_FORM_FETCHED,
+                    data: {
+                          form: pollCreationForm,
+                          emojis: emojis,
+                          gestures: gestures
+                    }
+                });
+
+            } catch (error) {
+                 return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+    /**
+     * Create poll
+     * @method
+     * @name create
+     * @param {Object} pollData - poll creation data
+     * @param {String} userId - userId
+     * @returns {String} - Sharable link .
+     */
+
+    static create(pollData= {}, userId= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (Object.keys(pollData).length == 0) {
+                    throw new Error (messageConstants.apiResponses.POLL_DATA_REQUIRED);
+                }
+
+                if (userId == "") {
+                    throw new Error (messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                let pollDocument = {
+                    name: pollData.name,
+                    creator: pollData.creator,
+                    createdBy: userId,
+                    startDate: new Date(),
+                    endDate: new Date(new Date().setDate(new Date().getDate() + pollData.endDate)),
+                    isDeleted: false,
+                    status: "active"
+                }
+                
+                let questionArray = [];
+                pollData.questions.forEach ( question => {
+                    
+                    let options = [];
+                    question.qid = gen.utils.uuidGenerate();
+                    
+                    if (question.options.length > 0) {
+                        let i = 1;
+                        question.options.forEach( option => {
+                            options.push(
+                               {
+                                 value: "R" + i,
+                                 label: option.label,
+                                 unicode: option.value
+                               }
+                           )
+
+                           ++i
+                        })
+                    }
+
+                    question.options = options;
+                    questionArray.push(question);
+                });
+                
+                pollDocument.questions = questionArray;
+                
+                let createPollResult = await database.models.polls.create(pollDocument)
+
+                let uuid = gen.utils.uuidGenerate();
+                let link = md5(uuid + "###" + createPollResult._id);
```

Comment:
@deepakr123 - Create utils function for md5

Reply:
resolved